src/js/angular/resources/controllers.js

```diff
@@ -99,12 +128,114 @@ resourcesCtrl.controller('ResourcesCtrl', ['$scope', '$timeout', 'MonitoringRest
                 $scope.loader = false;
             });
         };
+        let queryMonitorData;
+
+        const getQueryMonitor = function() {
+            const activeRepository = $scope.getActiveRepository();
+            if ($scope.error) {
+                return;
+            } else if (!activeRepository) {
+                queryMonitorData = $timeout(getQueryMonitor, POLLING_INTERVAL);
+                return;
+            }
+            $q.all([getPerformanceMonitorData(), getActiveQueryMonitor()])
```

Comment:
I'm not quite sure why you need to use the $q library for this? Doesn't Promise.all work here?

Reply:
There is a problem with Promise API and angularJs